### ISSUE TITLE
schema based validation for request data

### ISSUE THREAD

**Issue by marclave** [scalar/scalar#1778]
### What happens?

we currently dont format any types in the inputs based on the spec [https://swagger.io/docs/specification/data-models/data-types/](https://swagger.io/docs/specification/data-models/data-types/)

### Required Changes:

* When a specification file is loaded we save any request schemas to the workspace and reference them
* When entering request data the inputs should be validated against the schemas

### Considerations:

* When creating a new request how do we define the schema? (link vs direct entering)
* How do we handle no schema (likely just fallback to an "any" schema that we always have in the workspace)

**Comment by hanspagel** [comment]
Sorry, I’m missing some context. If someone passes `strong` (instead of `string`) we throw an error? Or show a warning?

**Comment by marclave** [comment]
> Sorry, I’m missing some context. If someone passes `strong` (instead of `string`) we throw an error? Or show a warning?

yeah inside of the openapi client inputs we can do some form of validation based on the formats :)

**Comment by AdeelK93** [comment]
This would be very helpful! React json schema form uses [ajv](https://github.com/ajv-validator/ajv) for this, maybe that library would be helpful here too?
